ios/Classes/OnePlugin.m

```diff
@@ -81,12 +85,54 @@ - (void)sendInteraction:(FlutterMethodCall*)call result:(FlutterResult) result
                                            details:error.localizedDescription]);
             } else {
                 [One processResponse:response];
-                result(response[@"tid"]);
+                result(response);
             }
         }];
     }
 }
 
+- (void)sendResponseCode:(FlutterMethodCall*)call result:(FlutterResult) result
+{
+    NSString *responseCode = call.arguments[@"responseCode"];
+    NSString *interactionPath = call.arguments[@"interactionPath"];
+    
+    if (!responseCode.length) {
+        return;
+    }
+
+    [One sendResponseCode:responseCode forInteractionPath:interactionPath];
+}
+
+- (void)optOut:(FlutterMethodCall *)call result:(FlutterResult) result
+{
+    NSNumber *optOutValue = call.arguments[@"optOut"];
+    BOOL optOut = optOutValue.boolValue;
+    NSArray *options = call.arguments[@"options"];
+
+    if (![options isKindOfClass:[NSNull class]] && [options isKindOfClass:[NSArray class]]) {
+        if (options.count) {
+            for (NSString *option in options) {
```

Comment:
similar problem here as mentioned in the android section. the last option in the list overrides all previous calls to opt out. take the list and build your options to pass to `[One opt]`

Reply:
because iOS behaves differently than android in that it can be configured piecewise, we should leave this as is.  let me know if you have objections.

Reply:
PR updated.  we now have a separate `optOutCityCountryDetection` api method to avoid confusion.  

Updated iOS so it is not piecewise to match android flow.
Calling the `optOutCityCountryDetection` will always opt the user back in (wipe and replace)